src/Components/ImportModal/ImportModal.tsx

```diff
@@ -5,19 +5,22 @@ import { DragModal, ImportComponents } from "./MediaPreviewer";
 import styles from "./ImportModal.module.css";
 import { Button, Container } from "react-bootstrap";
 import type { MediaPresenter } from "../MediaPresenter";
+import { observer } from 'mobx-react';
 
 type Props = {
   setShow: (show: boolean) => void;
   mediaPresenter: MediaPresenter;
 };
 
-export const ImportModal: React.FC<Props> = ({
+export const ImportModal: React.FC<Props> = observer(({
   setShow,
   mediaPresenter,
 }) => {
   const [onDragState, setOnDragState] = useState<boolean>(false);
+  // drop state can be deleted and check files length instead
   const [onDropState, setOnDropState] = useState<boolean>(false);
   const [mediaReady, setMediaReady] = useState<number>(0);
```

Comment:
As a follow-up: this can probably also go into the media presenter, and then you have a computed value `isMediaReady` which returns true when the media is ready

Reply:
This is a good idea, especially with the new UI :D Will do this soon